Dockerfile

```diff
@@ -2,23 +2,18 @@ FROM python:3.11-slim-bookworm AS build
 
 WORKDIR /opt/CTFd
 
-# hadolint ignore=DL3008
-RUN apt-get update \
-    && apt-get install -y --no-install-recommends \
-        build-essential \
-        libffi-dev \
-        libssl-dev \
-        git \
-    && apt-get clean \
-    && rm -rf /var/lib/apt/lists/* \
-    && python -m venv /opt/venv
+RUN python -m venv /opt/venv
 
 ENV PATH="/opt/venv/bin:$PATH"
 
-COPY . /opt/CTFd
+# Install CTFd requirements before copying the whole project,
+# this layer will be reused when rebuilding CTFd after a small change
+COPY ./requirements.txt /opt/CTFd
+RUN pip install --no-cache-dir -r requirements.txt
 
-RUN pip install --no-cache-dir -r requirements.txt \
-    && for d in CTFd/plugins/*; do \
+# Install CTFd plugins requirements
+COPY . /opt/CTFd
+RUN for d in CTFd/plugins/*; do \
```

Comment:
I think I would accept just this change. 

This first stage is just a build stage so while it is slow, it should not be frequently re-run. 

I don't want to risk breaking some kind of downstream plugin so I would rather keep these apt dependencies.

Reply:
I agree with this option. Do you prefer to cherry-pick 01ecab3e78dfc74ed17776dcd6c1b740dc30f729 or should I remove the other commit?

Reply:
If you want you can force push just the changes from 01ecab3 but commits get squashed down so it's not a big deal. Whichever is easiest for you.